infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -80,11 +80,17 @@ public CorfuServerNode(@Nonnull ServerContext serverContext) {
      * @param serverMap     Server Map with all components.
      */
     public CorfuServerNode(@Nonnull ServerContext serverContext,
-                           @Nonnull Map<Class, AbstractServer> serverMap) {
+                           @Nonnull ImmutableMap<Class, AbstractServer> serverMap) {
         this.serverContext = serverContext;
         this.serverMap = serverMap;
-        router = new NettyServerRouter(new ArrayList<>(serverMap.values()));
+        router = new NettyServerRouter(serverMap.values().asList(), serverContext);
         this.serverContext.setServerRouter(router);
+        // If the node is started in the single node setup and was bootstrapped,
+        // set the server epoch as well.
+        if(serverContext.isSingleNodeSetup() && serverContext.getCurrentLayout() != null){
```

Comment:
Can you explain the necessity of this block?

Reply:
ServerContext is now initialized with the invalid epoch. In the normal node to set it to the first epoch (0), we bootstrap it with a layout and seal the epoch from the client. After this happens, the client can start routing messages to the node safely. In the single node, we don't bootstrap it since the node self-bootstraps on a startup from the layout server. This block ensures that we self-seal the node in the single node too after everything has been initialized. Otherwise, in the single node setup the log unit seal epoch would be invalid, and all the messages won't be able to go through.